b2share/modules/records/minters.py

```diff
@@ -41,111 +41,110 @@
 
 def b2share_record_uuid_minter(record_uuid, data):
     """Use record's UUID as unique identifier."""
-    provider = RecordUUIDProvider.create(
-        object_type='rec', object_uuid=record_uuid,
+    rec_pid = RecordUUIDProvider.create(
+        object_type='rec',
+        object_uuid=record_uuid,
         pid_value=data['_deposit']['id']
-    )
+    ).pid
     if '_pid' not in data:
         data['_pid'] = []
     data['_pid'].append({
-        'value': provider.pid.pid_value,
-        'type': provider.pid.pid_type,
+        'value': rec_pid.pid_value,
+        'type': rec_pid.pid_type,
     })
 
-    b2share_oaiid_minter(record_uuid, data)
-    b2share_pid_minter(record_uuid, data)
-    if current_app.config.get('AUTOMATICALLY_ASSIGN_DOI'):
-        b2share_doi_minter(record_uuid, data)
+    b2share_oaiid_minter(rec_pid, data)
+    b2share_pid_minter(rec_pid, data)
+    if current_app.config.get('TESTING', False) or current_app.config.get('FAKE_DOI', False):
+        b2share_doi_minter(rec_pid, data, fake_it=True)
+    elif current_app.config.get('AUTOMATICALLY_ASSIGN_DOI'):
+        b2share_doi_minter(rec_pid, data)
```

Comment:
So in the case FAKE_DOI=True and AUTOMATICALLY_ASSIGN_DOI=False we assign (fake) DOI. Does that make sense?